[WFLY-12985] Provide a separate feature pack and dist for non-observability MP specs

bstansberry opened the pull request:
https://issues.redhat.com/browse/WFLY-12985

bstansberry committed:
[WFLY-12985] Rename galleon-pack, build and dist maven modules by prefixing with 'ee-'

This is a foundational step, putting in place a structure for producing our traditional top-level feature pack and dists.

bstansberry committed:
[WFLY-12985] Create new galleon-pack, build and dist maven modules to hold MP content

Create the new feature pack and dists. These use the traditional maven module names and GAVs as our standard dist will include the rapidly evolving MP items. New artifact ids are given to the non-MP feature pack and dist artifacts.

bstansberry committed:
[WFLY-12985] Use a distinct product module for the ee dist vs the complete one

bstansberry committed:
[WFLY-12985] Facilitate separate maven versions for ee dist vs complete dist

This may be useful for EAP.

bstansberry committed:
[WFLY-12985] Move fault-tolerance from the ee feature pack and dists to the full one

This removes fault-tolerance from the 'legacy' non-Galleon feature packs. Legacy f-ps are only maintained to help Keycloak (which doesn't need MP) until it can move to Galleon.

This requires overrides of the Galleon domain config and the high level feature groups used in it in order to integrate the MP variant of domain-host-excludes.

bstansberry committed:
[WFLY-12985] Move openapi from the ee feature pack and dists to the full one

As with Fault Tolerance this means removal of this feature from the legacy non-Galleon feature pack.

bstansberry committed:
[WFLY-12985] Move jwt from the ee feature pack and dists to the full one

As with FaultTolerance and OpenAPI this means removal of the feature from the legacy non-Galleon feature pack.

Unlike FaultTolerance and OpenAPI, JWT is meant to be part of our standard configs, so this requires overrides of the various Galleon configs for the standalone model, plus the high level feature-groups used for those models and for domain.xml

bstansberry commented:
retest this please

bstansberry commented:
retest this please

bstansberry commented:
@jamezp I fixed the -Drelease problem. Thanks for pointing it out!

jamezp reviewed (approved):
This looks good to me.

bstansberry committed:
[WFLY-12985] Allow test jobs to control which feature pack is used to provision 'ee' servers

Both the top level 'wildfly' feature pack and its dependency the 'wildfly-ee' feature pack should work. Default to testing using 'wildfly' but let test jobs override this to test the other option.

bstansberry commented:
retest this please

bstansberry committed:
[WFLY-12985] Allow test jobs to control which maven project is the source of the build under test

bstansberry commented:
retest this please

yersan commented on ee-galleon-pack/src/main/resources/packages/product.conf/content/modules/system/layers/base/org/jboss/as/product/wildfly-ee/module.xml:
```diff
@@ -0,0 +1,38 @@
+  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  -->
+
+<module xmlns="urn:jboss:module:1.5" name="org.jboss.as.product" slot="wildfly-ee">
```
Should not this change be also effective for the legacy product module.xml?
https://github.com/wildfly/wildfly/blob/master/dist-legacy/src/distribution/resources/modules/system/layers/base/org/jboss/as/product/wildfly-full/module.xml#L25

We could add and additional verification to ensure modules/system/layers/base/org/jboss/as/product/wildfly-ee/module.xml has the correct one

bstansberry replied:
I think it should, yes.

bstansberry replied:
Oh, definitely it should, as the name of the module dir was not matching the module.xml!

bstansberry commented on ee-galleon-pack/pom.xml:
```diff
@@ -0,0 +1,4155 @@
+
+        <dependency>
+            <groupId>org.eclipse.microprofile.config</groupId>
+            <artifactId>microprofile-config-api<
```
Thanks; it looks like I missed deleting this one. I'll update the relevant commit.

bstansberry committed:
[WFLY-12985] Support using -Ddisable.microprofile.tests to turn off surefire executions that must run against a dist with the full set of MP functionality.

If a test job is configured using jboss.dist to point at an external distribution and that distribution doesn't have the full set of MP features, use this ability to turn off parts of the testsuite that are expected to fail.

bstansberry committed:
[WFLY-12985] Separate the layers in the wildfly-ee feature pack in the Galleon layers doc page

Also, WFLY-13027 -- add a doc line for microprofile-openapi

jfdenise commented on docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc:
```diff
@@ -115,6 +115,14 @@ WildFly defined layers to assemble a custom configuration. <<galleon-cli-recipes
 Basic layers:
 
+* _microprofile-fault-tolerance_: Support for MicroProfile Fault Tolerance.
+* _microprofile-jwt_: Support for MicroProfile JWT.
```
These layers are still documented in the WildFly EE layers, should be removed from EE.

bstansberry replied:
Thanks; will fix.

bstansberry replied:
Fixed.

dosoudil commented on feature-pack/pom.xml:
```diff
@@ -332,17 +332,6 @@
             <artifactId>jakarta.security.enterprise</artifactId>
         </dependency>
 
-        <dependency>
```
If these artifacts are removed from feature-pack, they're not runtime artifacts anymore.
Moving them to galleon-pack means that they're build dependencies, not runtime.
Is it correct from the perspective of using WildFly Server?
cc @rgdoliveira

bstansberry replied:
They are still runtime dependencies of any server that uses the 'wildfly' galleon pack. The output of this maven module and the 'servlet-feature-pack' maven module are 'legacy feature packs' and are not used in any WildFly distribution. We just maintain these modules so projects that still want to create custom WF-technology-based servers using the legacy feature packs can use their output. But an official WildFly dist does not use the output of these. (You could do 'mvn install -pl dist -am' and this maven module wouldn't even build.) 

However, our official dists do use some *source* from these maven modules. This is so we can build the official Galleon feature packs and these legacy ones and not have to duplicate the source, which would almost certainly lead to bugs.  The Galleon feature pack poms actually copy over the source resources they need:

https://github.com/wildfly/wildfly/blob/master/galleon-pack/pom.xml#L73
https://github.com/wildfly/wildfly/blob/master/servlet-galleon-pack/pom.xml#L548

bstansberry commented:
retest this please

bstansberry commented:
retest this please
